src/AndersonAcceleration.f90

```diff
@@ -251,23 +261,33 @@ SUBROUTINE step_AndersonAccelerationType(solver,x_new)
       solver%alpha(depth_s+1)=solver%alpha(depth_s+1)-solver%alpha(i)
     ENDDO
 
-    !Get accelerated solution
-    x_new(:)=0.0_SRK
-    DO i=1,depth_s+1
-      x_new(:)=x_new(:)+solver%alpha(i)*(solver%x(:,i)+solver%beta*solver%r(:,i))
-    ENDDO
+    !Ensure Anderson coefficient solve did not fail due to bad input vector
+    IF(ISNAN(solver%alpha(depth_s+1))) THEN
+      !Bad Anderson coefficient solve, revert to under-relaxation and reset Anderson solver
```

Comment:
should we make sure we don't drop into some infinite loop where we keep failing?

Reply:
We shouldn't end up in an infinite loop..."shouldn't". I've made the behavior here when Anderson fails to just do simple under-relaxation (like default) using the mixing parameter, and then that call to reset will throw away all the past solutions (that would have caused the failure). If we ran into a case where it would fail over and over it would just act like default.

Reply:
I need to add a test for this ability in the unit test.

Reply:
Improved the rescue to look for when Alpha = 0.0 this signals that a duplicate or other bad iterate has been handed in and added testing.